Clarification and details on CoC and Contribution guideline: Request for comments (#6)
virtual-biohackathons/covid-19-bh20: CONTRIBUTING.md

# How to contribute?

First of all, thanks for taking the time to contribute! :tada::+1:

Your contributions are important to make this project and event successful. 
You can report mistakes and errors, propose a project, offer dataset and resources or create more content. 

Whatever is your background, there is a way to contribute on this [GitHub repository](https://github.com/virtual-biohackathons/covid-19-bh20), 
on our [mailing list](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!forum/virtual-biohackathon)

> or other [related projects listed on our wiki page](https://github.com/virtual-biohackathons/covid-19-bh20/wiki).

Reviewer: ```suggestion
, other [related projects listed on our wiki page](https://github.com/virtual-biohackathons/covid-19-bh20/wiki) or via [slack conversations](https://virtualbiohac-xt62674.slack.com/join/shared_invite/zt-cuur40oj-wdrVz50NocwVrH7vgKTdPg).
```

You can raise any issue relevant to this project/event publicly in the mailing list (google group), we will address and/or assess your change proposal as promptly as we
can to make your participation easier.

We have a [Code of Conduct](https://github.com/virtual-biohackathons/covid-19-bh20/blob/master/CODE_OF_CONDUCT.md) that applies to all the activities and projects involved in this event.
For any organisation related queries or concern, you can directly reach out to the organisers Leyla Garcia (email: ljgarciaco@gmail.com), Pjotr Prins (email: pjotr.public708@thebird.nl) 
and  Tazro Ohta (email: t.ohta@dbcls.rois.ac.jp). 

> **ADD MISSING DETAILS**

Reviewer: ```suggestion
- With such a large group of participants, we are encouraging a bottom-up approach, where groups are formed around projects/topics and work (kind of) independently from others. 
- If you need any information from another group/topic, you can always reach people via email or slack (please note that as the hacking starts, Slack might become more interactive, you can also directly ping people there, be patient please and re-ping if needed or look for an alternative contact). 
- Every group/topic will have leader/coordinator/facilitator, please check the wiki to get more information (as we get close to the hacking week, we expect the wiki to change quickly and people to add more group/topic details there). 
- We also expect group/topic leaders/coordinators/facilitators to provide a quick (daily if possible) summary via the wiki so everybody can get an idea on it.
```

> ## How can I participate in this project/event?

Reviewer: ```suggestion
## How can I participate at this BioHackathon community during and after the event?
```

> **ADD DETAILS**

Reviewer: ```suggestion
Join the [Google group mailing list](https://groups.google.com/forum/#!forum/virtual-biohackathon) and the [virtualbiohackathoncovid19 slack](https://join.slack.com/t/virtualbiohac-xt62674/shared_invite/zt-cuur40oj-wdrVz50NocwVrH7vgKTdPg) and join the conversation! No approval needed. 

Please have a look at the [GitHub wiki](https://github.com/virtual-biohackathons/covid-19-bh20/wiki) to see the existing projects. 
This will help you identify possible ways to contribute to. 
Contact, via the mailing list or slack, the community working on that project and let them know of your interest and how you would like to participate. Conversation and collaboration will start from it.
```

## How can I fix something on the repository?

- Raise mistakes, error or missing information on this repository by opening a Pull Request
  - Submit trivial fixes (for example, a typo, a broken link or an obvious error)
  - Start work on a contribution that was already asked for, or that you’ve already discussed, in a mailing list
  - A pull request doesn’t have to represent finished work. It’s usually better to open a pull request early on, so others can watch or give feedback on your progress. Just mark it as a “WIP” (Work in Progress) in the subject line. You can always add more commits later.
  - [Read details for open a Pull request](https://opensource.guide/how-to-contribute/#opening-a-pull-request)

## How can I join the mailing list and participate in an ongoing conversation?

> **ADD DETAILS**

Reviewer: ```suggestion
The mailing list works via a Google group, so by [joining the group](https://groups.google.com/forum/#!forum/virtual-biohackathon) you will get new messages and will be allowed to post.

We also have a [slack]((https://virtualbiohac-xt62674.slack.com/join/shared_invite/zt-cuur40oj-wdrVz50NocwVrH7vgKTdPg)) where channels (commonly corresponding to groups/topics/projects) are created and people can join and contribute to the discussion.
```

## How can join a project during the hackthon?
- When you’ve found a project you’d like to contribute to, 
do a quick scan to make sure that the project is suitable for accepting contributions. 

> **ADD DETAILS**

Reviewer: ```suggestion
The earlier you join a group the better. Commonly, groups/topics are created on slack, so please have a look there to see if the group/topic you are interested in is already there. If not, go to general or COVID-19 and raise a question on how to join that group/topic/project. Someone will point you in the right direction. 
```

## How can I offer dataset and other compute resources?

> **ADD DETAILS**

Reviewer: ```suggestion
People are pointing to data or offering compute resources via email. A dedicated wiki entry will include more details on the known/available data and compute resources.
```

  - [Read details for open a Pull request](https://opensource.guide/how-to-contribute/#opening-a-pull-request)

## How can I join the mailing list and participate in an ongoing conversation?

The mailing list works via a Google group, so by [joining the group](https://groups.google.com/forum/#!forum/virtual-biohackathon) you will get new messages and will be allowed to post.

We also have a [slack]((https://virtualbiohac-xt62674.slack.com/join/shared_invite/zt-cuur40oj-wdrVz50NocwVrH7vgKTdPg)) where channels (commonly corresponding to groups/topics/projects) are created and people can join and contribute to the discussion.

## How can I offer dataset and other compute resources? What are the data protection policy?

People are pointing to data or offering compute resources via email. A dedicated wiki entry will include more details on the known/available data and compute resources.

## How can I use dataset and other compute resources? 

> **ADD DETAILS**

Reviewer: ```suggestion
We have community-based list of available [datasets and tools](./datasets_and_tools.md) as well as a list of [computing resources](./compute_resources.md). Please visit those pages for more information.
```

## What data protection policies and licenses are applied?

> **ADD DETAILS**

Reviewer: ```suggestion
- As different datasets, tools and resources will be used, we, therefore, encourage you to double-check for any limitations the data and tools you are using. Wherever possible please document those limitations and how you are troubleshooting those issues within your team. 
- A lot of freely available and open access initiatives have been put in place as a way to contribute to solutions around COVID-19. If the dataset/tool you want to use has any restrictions, please contact the responsible team/owner. If the owners can not relax their regular policies/licenses for the use in this hackathon, please find an alternative data/tool to use. Make sure that you document these restrictions to be shared with your team members and anyone who would like to join your team during the hackathon or use your work afterwards.
```

## Are there more ways to participate?

> **ADD DETAILS**

Reviewer: ```suggestion
Other than joining the conversation and contributing to any of the topics, you can also offer additional compute support, or data or tools. For instance, [ELIXIR Europe](https://elixir-europe.org/) and [Galaxy](https://galaxyproject.org/) have kindly agreed to contribute with cloud resources, servers and others.  See the section above related to datasets and compute resources.
```

- If you need any information from another group/topic, you can always reach people via email or slack (please note that as the hacking starts, Slack might become more interactive, you can also directly ping people there, be patient please and re-ping if needed or look for an alternative contact). 
- Every group/topic will have leader/coordinator/facilitator, please check the wiki to get more information (as we get close to the hacking week, we expect the wiki to change quickly and people to add more group/topic details there). 
- We also expect group/topic leaders/coordinators/facilitators to provide a quick (daily if possible) summary via the wiki so everybody can get an idea on it.

## How can I participate at this event?

Join the [Google group mailing list](https://groups.google.com/forum/#!forum/virtual-biohackathon) and the [virtualbiohackathoncovid19 slack](https://join.slack.com/t/virtualbiohac-xt62674/shared_invite/zt-cuur40oj-wdrVz50NocwVrH7vgKTdPg) and join the conversation! No approval needed. 

> ## How can join the hackthon?

Reviewer 2: ```suggestion
## How can I join the hackathon?
```

- Please have a look at the [GitHub wiki](https://github.com/virtual-biohackathons/covid-19-bh20/wiki) to see the existing projects. 
- When you’ve found a project you’d like to contribute to, 
do a quick scan to make sure that the project is suitable for accepting contributions.
- Commonly, channels are created on our Slack workspace, so please have a look there to see if the group/topic you are interested in is already there. 

> - If there is no channel, go to general or COVID-19 mailing list and raise a question on how to join that group/topic/project. Members working on that project would help you participate.

Reviewer 2: Re: _general or COVID-19 mailing list_, isn't there just one, and isn't this the Google Group? Or are there others that I missed? Either way, this should probably be linked to what is the right answer?

Reviewer: Instead of 
- If there is no channel, go to general or COVID-19 mailing list and raise a question on how to join that group/topic/project. Members working on that project would help you participate.
It should be
- If there is no channel, go to "general" slack channel or [COVID-19 Google group mailing list](https://groups.google.com/forum/#!forum/virtual-biohackathon) and raise a question on how to join that group/topic/project. Members working on that project would help you participate.

Reviewer: @malvikasharan I tried to create a patch with this change (on your forked copy) but I got a GitHub error. Could you please add this change? Once that is done, I will accept and merge this PR. Thanks.

## How can I fix something on the repository?

> - Raise mistakes, error or missing information on this repository by opening a Pull Request

Reviewer 2: It looks like Issues have been disabled for this repo. Given this statement, this would seem to be intentional; i.e., filing an issue is not a good way to contribute here. That's acceptable (even if not what I would have chosen), just trying to make sure that this is indeed on purpose?

Reviewer: As far as I understand, it is

  - Submit trivial fixes (for example, a typo, a broken link or an obvious error)
  - Start work on a contribution that was already asked for, or that you’ve already discussed, in a mailing list
  - A pull request doesn’t have to represent finished work. It’s usually better to open a pull request early on, so others can watch or give feedback on your progress. Just mark it as a “WIP” (Work in Progress) in the subject line. You can always add more commits later.
  - [Read details for open a Pull request](https://opensource.guide/how-to-contribute/#opening-a-pull-request)

## How can I join the mailing list and participate in an ongoing conversation?

The mailing list works via a Google group, so by [joining the group](https://groups.google.com/forum/#!forum/virtual-biohackathon) you will get new messages and will be allowed to post.

We also have a [slack]((https://virtualbiohac-xt62674.slack.com/join/shared_invite/zt-cuur40oj-wdrVz50NocwVrH7vgKTdPg)) where channels (commonly corresponding to groups/topics/projects) are created and people can join and contribute to the discussion.

> ## How can I offer dataset and other compute resources? What are the data protection policy?

Reviewer 2: ```suggestion
## How can I offer dataset and other compute resources? What are the data protection policies?
```
